tests/foreman/longrun/test_oscap.py

```diff
@@ -493,3 +493,131 @@ def test_positive_reporting_emails_of_oscap_reports():
+        4. Associate the policy with a hostgroup
+        5. Run the Ansible job and then trigger the Oscap job.
+        6. Oscap must Utilize the local files for the client scan.
+
+    :expectedresults: Oscap run should happen using the --localfile argument.
+
+    :BZ: 2081777,2211952
+
+    :CaseImportance: Critical
+    """
+    SELECTED_ROLE = 'theforeman.foreman_scap_client'
+    file_name = 'security-data-oval-com.redhat.rhsa-RHEL8.xml.bz2'
+    download_url = 'https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml.bz2'
+    profile = OSCAP_PROFILE['ospp8']
+    content = OSCAP_DEFAULT_CONTENT[f'{distro}_content']
+    hgrp_name = gen_string('alpha')
+    policy_name = gen_string('alpha')
+
+    target_sat.cli_factory.make_hostgroup(
+        {
+            'content-source-id': default_proxy,
+            'name': hgrp_name,
+            'organizations': module_org.name,
+        }
+    )
+    # Creates oscap_policy.
+    scap_id, scap_profile_id = fetch_scap_and_profile_id(content, profile)
+    with Broker(
+        nick=distro,
+        host_class=ContentHost,
+        target_cores=OSCAP_TARGET_CORES,
+        target_memory=OSCAP_TARGET_MEMORY,
+    ) as vm:
+        host_name, _, host_domain = vm.hostname.partition('.')
+        vm.create_custom_repos(
+            **{
+                'baseos': settings.repos.rhel8_os.baseos,
+                'appstream': settings.repos.rhel8_os.appstream,
+            }
+        )
+        vm.create_custom_repos(
+            sat_client=settings.repos['SATCLIENT_REPO']['RHEL8'],
+        )
+
+        result = vm.register(module_org, None, ak_name[distro], target_sat)
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        proxy_id = target_sat.nailgun_smart_proxy.id
+        target_host = vm.nailgun_host
+        target_sat.api.AnsibleRoles().sync(
+            data={'proxy_id': proxy_id, 'role_names': [SELECTED_ROLE]}
+        )
+        role_id = (
+            target_sat.api.AnsibleRoles().search(query={'search': f'name={SELECTED_ROLE}'})[0].id
+        )
+        target_sat.api.Host(id=target_host.id).add_ansible_role(data={'ansible_role_id': role_id})
+        host_roles = target_host.list_ansible_roles()
+        assert host_roles[0]['name'] == SELECTED_ROLE
+        target_sat.cli_factory.make_scap_policy(
+            {
+                'scap-content-id': scap_id,
+                'hostgroups': hgrp_name,
+                'deploy-by': 'ansible',
+                'name': policy_name,
+                'period': OSCAP_PERIOD['weekly'].lower(),
+                'scap-content-profile-id': scap_profile_id,
+                'weekday': OSCAP_WEEKDAY['friday'].lower(),
+                'organizations': module_org.name,
+            }
+        )
+
+        vm.execute(f'curl -o {file_name} {download_url}')
+        target_sat.cli.Host.update(
+            {
+                'name': vm.hostname.lower(),
```

Comment:
hostnames should be lowercase only 
```suggestion
                'name': vm.hostname,
```